crates/ide/src/inlay_hints/binding_mode.rs

```diff
@@ -148,7 +148,6 @@ struct Struct {
     field: &'static str,
 }
 fn foo(s @ Struct { field, .. }: &Struct) {}
-     //^^^^^^^^^^^^^^^^^^^^^^^^ref
```

Comment:
I think `s` here is binding by moving the reference itself, not binding by ref the thing inside of reference. This example makes the difference clear:

```Rust
let t @ T = &&&T; // type is `&&&T`, binding mode is `move`
let (t @ T,) = &&&(T,); // type is `&T`, binding mode is `ref`
```

Reply:
Aha, turns out I completely misunderstood how binding modes worked for @ bindings! I *think* you are right here (at least judging from testing with `==` to avoid coercions from messing things up).

That also explains the pattern type mismatches now. Thanks for clarifying